src/main/scala/karazin/scala/users/group/week2/homework/Homework.scala

```diff
@@ -35,39 +35,39 @@ object Homework:
 
     @targetName("addition")
     infix def +(that: Rational): Rational =
-      Rational(this.numer*that.denom+that.numer*this.denom,this.denom*that.denom)
+      Rational(this.numer * that.denom + that.numer * this.denom, this.denom * that.denom)
 
     @targetName("negation")
     infix def unary_- : Rational =
       -this
 
     @targetName("substraction")
     infix def -(that: Rational): Rational =
-      Rational(this.numer*that.denom-that.numer*this.denom,this.denom*that.denom)
+      Rational(this.numer * that.denom - that.numer * this.denom, this.denom * that.denom)
 
     @targetName("multiplication")
     infix def *(that: Rational): Rational =
-      Rational(this.numer*that.numer,this.denom*that.denom)
+      Rational(this.numer * that.numer, this.denom * that.denom)
 
     @targetName("devision")
     infix def /(that: Rational): Rational =
-      Rational(this.numer*that.denom,this.denom*that.numer)
+      Rational(this.numer * that.denom, this.denom * that.numer)
 
     override def toString: String = s"${this.numer}/${this.denom}"
 
+    @tailrec
     private def gcd(a: Int, b: Int): Int =
       if b == 0 then a else gcd(b, a % b)
 
     private lazy val g = gcd(abs(x), y)
 
     override def equals(other: Any): Boolean =
```

Comment:
Contract violation: https://www.baeldung.com/java-equals-hashcode-contracts
If objects are equal by `equals` then they should have the same `hashCode`.